### PR TITLE
better handling of redirects to install complete step

### PR DESCRIPTION
**Pull request by alex-kovoy**
Instead of redirecting to last installation step (bandwagon app), redirect a user back to installer if last installation step has not been completed.

The installer in this case will show that installation has been completed and ask a user to click a button to complete the installation.

Also fixes https://github.com/gravitational/gravity.e/issues/3781

### CODE REVIEW

File: lib/webapi/webhandler.go [hunk 1]
```diff
@@ -101,40 +99,29 @@ func NewHandler(cfg WebHandlerConfig) *WebHandler {
...
-		"/web/site/:site_domain",
-		"/web/site/:site_domain/*rest",
-	}
-	for _, route := range siteRoutes {
-		for _, m := range []string{"GET", "POST"} {
-			wh.Handle(m, route, wh.needsLogin(wh.siteHandler))
-		}
-	}
+	// installer
+	wh.Handle("GET", "/web/installer/new/*rest", wh.needsLogin(wh.installerHandler))
```
Comment (r0mant) [inline lib/webapi/webhandler.go]: What about post, put and other types of requests that were handled before, here and below?

File: lib/webapi/webhandler.go [hunk 2]
```diff
@@ -196,6 +183,21 @@ func (h *WebHandler) configHandler(w http.ResponseWriter, r *http.Request, p htt
 	fmt.Fprintf(w, "var GRV_CONFIG = %v;", string(out))
 }
 
+func (h *WebHandler) rootHandler(w http.ResponseWriter, r *http.Request, p httprouter.Params, s session) {
+	site, err := h.cfg.Operator.GetLocalSite()
```
Comment (r0mant) [inline lib/webapi/webhandler.go]: site -> cluster

File: lib/webapi/webhandler.go [hunk 3]
```diff
@@ -196,6 +183,21 @@ func (h *WebHandler) configHandler(w http.ResponseWriter, r *http.Request, p htt
...
+	if err != nil && !trace.IsNotFound(err) {
+		log.Errorf("failed to get local site: %v", trace.DebugReport(err))
```
Comment (r0mant) [inline lib/webapi/webhandler.go]: Capitalize "Failed" plz.

File: lib/webapi/webhandler.go [hunk 4]
```diff
@@ -307,8 +309,8 @@ func (h *WebHandler) siteHandler(w http.ResponseWriter, r *http.Request, p httpr
 	}
 
 	// if the app defines custom install step and the site hasn't completed it yet,
-	// always redirect the site to it
-	http.Redirect(w, r, "/web/installer/site/"+siteDomain+"/complete/", http.StatusFound)
+	// always redirect the site back to installer
+	http.Redirect(w, r, "/web/installer/site/"+siteDomain, http.StatusFound)
```
Comment (r0mant) [inline lib/webapi/webhandler.go]: This means the user will see "installation complete" page again and will have to click "continue", right? Would it be complicated to fix this session storage issue you told me about and redirect directly to bandwagon instead?

File: lib/webapi/webhandler.go (hunk 1 above)
Comment (alex-kovoy) [reply lib/webapi/webhandler.go]: there are no need to have them as there is no exposed APIs using these verbs

File: lib/webapi/webhandler.go (hunk 4 above)
Comment (alex-kovoy) [reply lib/webapi/webhandler.go]: This approach is simpler as we do not have much control over customized bandwagon step

File: lib/webapi/webhandler.go (hunk 1 above)
Comment (r0mant) [reply lib/webapi/webhandler.go]: So when bandwagon app does a POST request - how is this handled then?

Comment (alex-kovoy) [reply lib/webapi/webhandler.go]: good catch

**Pull request merged**